sdk/pinocchio/src/sysvars/mod.rs

```diff
@@ -44,3 +44,48 @@ macro_rules! impl_sysvar_get {
         }
     };
 }
+
+/// Handler for retrieving a slice of sysvar data from the `sol_get_sysvar`
+/// syscall.
+///
+/// # Safety
+///
+/// `dst` must point to a buffer that can hold at least `length` bytes.
+#[inline]
+pub unsafe fn get_sysvar_unchecked(
+    dst: *mut u8,
+    sysvar_id: &Pubkey,
+    offset: u64,
+    length: u64,
+) -> Result<(), ProgramError> {
+    let sysvar_id = sysvar_id as *const _ as *const u8;
+
+    #[cfg(target_os = "solana")]
+    let result = unsafe { crate::syscalls::sol_get_sysvar(sysvar_id, dst, offset, length) };
```

Comment:
As `get_sysvar_unchecked()` is marked `unsafe` this inner `unsafe` block has no meaning, I think.

```suggestion
    let result = crate::syscalls::sol_get_sysvar(sysvar_id, dst, offset, length);
```

Reply:
It will be a warning in [rust 2024](https://doc.rust-lang.org/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html)